tests/test_migrations.py

```
@@ -0,0 +1,41 @@
+            self.assertEqual(versions, [migration.version for migration in MIGRATIONS])
+
+    def test_legacy_database_receives_missing_columns(self):
+        with tempfile.TemporaryDirectory() as temporary:
+            path = Path(temporary) / "catalog.db"
+            conn = sqlite3.connect(path)
+            conn.executescript("""
+                CREATE TABLE roots(id INTEGER PRIMARY KEY,path TEXT,kind TEXT,label TEXT,enabled INTEGER DEFAULT 1,last_scanned_at TEXT);
+                CREATE TABLE titles(id INTEGER PRIMARY KEY,root_id INTEGER,kind TEXT,title TEXT,year INTEGER,folder_path TEXT,metadata_title TEXT,updated_at TEXT);
+                CREATE TABLE files(id INTEGER PRIMARY KEY,title_id INTEGER,path TEXT,filename TEXT,extension TEXT,size_bytes INTEGER,modified_at REAL,season INTEGER,episode_start INTEGER,episode_end INTEGER,parsed_title TEXT,seen_scan TEXT);
+            """)
+            conn.commit(); conn.close()
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)